sympy/sets/sets.py

```diff
@@ -1281,6 +1323,16 @@ def _measure(self):
             parity *= -1
         return measure
 
+    @property
+    def _kind(self):
+        args = simplify_union(self.args).args # for evalute = False
```

Comment:
This function should definitely not be called here. Checking a kind should be an efficient operation.

Reply:
Same goes for all the other cases where similar functions are called.

Reply:
`args = simplify_union(self.args).args` is used for case where  `evaluate = False`. I am not able to find any other way to get args of simplified union, so that both `Union({1}, EmptySet, evaluate=False).kind`, `Union({1}, EmptySet).kind` give same results

Reply:
You can just do `args = (arg for arg in args if arg is not EmptySet)`.